src/EnergyPlus/NodeInputManager.cc

```diff
@@ -867,13 +871,11 @@ void InitUniqueNodeCheck(EnergyPlusData &state, std::string const &ContextName)
     state.dataNodeInputMgr->CurCheckContextName = ContextName;
 }
 
-void CheckUniqueNodes(EnergyPlusData &state,
-                      std::string const &NodeTypes,
-                      std::string const &CheckType,
-                      bool &ErrorsFound,
-                      Optional_string_const CheckName,
-                      Optional_int_const CheckNumber,
-                      Optional_string_const ObjectName)
```

Comment:
This function was checking for uniqueness of an optional name or optional number based on the string in CheckType... which appears to always be one of two literal strings... so known at runtime. Replace that with two functions, one that checks for name and one that checks for number.